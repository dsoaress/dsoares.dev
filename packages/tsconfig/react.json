{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "React",
  "extends": "./base.json",
  "compilerOptions": {
    "jsx": "react-jsx",
    "lib": ["ES2015"],
    "module": "ESNext",
    "target": "es6"
  }
}
