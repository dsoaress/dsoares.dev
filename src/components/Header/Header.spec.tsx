import { render, screen } from '@testing-library/react'

import { Header } from '.'

const profile = { title: 'John Doe' }
const description = 'John Doe is a software engineer.'
const resume = { title: 'resume', file: '' }

jest.mock(
  'next/image',
  () =>
    function Image({ src, alt }: { src: string; alt: string }) {
      // eslint-disable-next-line @next/next/no-img-element
      return <img src={src} alt={alt} />
    }
)

jest.mock('@/components/Skills', () => ({
  Skills: () => <div data-testid="skills" />
}))

jest.mock('next/router', () => ({
  useRouter: () => ({ locale: 'en' })
}))

jest.mock('@/hooks/useData', () => ({
  useData: () => ({
    locale: 'en',
    c: {
      profile: { title: profile.title },
      showResume: true
    },
    t: {
      description,
      resume: { title: resume.title, file: resume.file }
    }
  })
}))

describe('<Header />', () => {
  let element: HTMLElement | null = null

  beforeEach(() => {
    const { container } = render(<Header avatarBlurDataURL="" />)
    element = container
  })

  it('should matches snapshot', () => {
    expect(element).toMatchSnapshot()
  })

  it('should render with correct title', () => {
    expect(screen.getByText(profile.title)).toBeInTheDocument()
  })

  it('should render with correct description', () => {
    expect(screen.getByText(description)).toBeInTheDocument()
  })

  it('should render with correct avatar', () => {
    expect(screen.getByAltText(profile.title)).toBeInTheDocument()
  })

  it('should render resume button', () => {
    expect(screen.getByText(resume.title)).toBeInTheDocument()
  })

  it('should render with skills component in the right place', () => {
    expect(screen.getByTestId('skills')).toBeInTheDocument()
  })
})
