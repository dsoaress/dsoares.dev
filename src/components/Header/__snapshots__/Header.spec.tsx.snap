// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should matches snapshot 1`] = `
<div>
  <header
    class="absolute flex h-screen w-full items-end justify-center pb-20 md:pb-8"
  >
    <div
      class="z-10 grid gap-8 p-8 text-center"
    >
      <svg
        class="fill-current text-primary-600 dark:text-primary-500 mx-auto hidden sm:block"
        clip-rule="evenodd"
        data-testid="logo"
        fill-rule="evenodd"
        height="80"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        viewBox="0 0 375 548"
        width="80"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M274.023 527.357L19.241 434.973V190.092l210.81-81.362a9.607 9.607 0 006.153-8.97V46.448l54.108-22.495v363.6l-56.103-20.487V174.298c0-3.168-1.57-6.14-4.185-7.933a9.565 9.565 0 00-8.91-1.03L84.926 218.153a9.618 9.618 0 00-6.14 8.97l.072 179.678a9.634 9.634 0 006.388 9.053l251.275 89.634-62.498 21.87zm-59.05-167.3l-43.21-15.767-.117-57.974 43.327-15.807v89.549zM355.776 491.93l-257.69-91.915-.065-166.315 116.946-45.362-.033 61.716-56.207 20.507a9.616 9.616 0 00-6.323 9.054l.15 71.421a9.617 9.617 0 006.316 9.015l137.759 50.276c.743.267 1.499.437 2.262.521a9.622 9.622 0 006.55-1.675 9.62 9.62 0 004.1-7.88V24.22l46.235 20.122V491.93zm19.23-453.885a9.614 9.614 0 00-5.783-8.82L303.766.736c-.104-.047-.221-.06-.326-.099-.384-.15-.775-.24-1.173-.339-.365-.097-.73-.202-1.102-.247-.404-.052-.801-.046-1.205-.046-.378 0-.75-.013-1.122.026-.41.046-.801.157-1.199.254-.365.085-.73.163-1.082.3-.11.033-.215.04-.32.085L222.9 31.149a9.616 9.616 0 00-5.925 8.878V93.17L6.158 174.526a9.62 9.62 0 00-6.153 8.97v258.217a9.62 9.62 0 006.343 9.04L270.686 546.6a9.623 9.623 0 003.278.574 9.671 9.671 0 003.181-.535l91.42-31.985c.45-.163.821-.45 1.239-.671.371-.19.775-.313 1.127-.555.046-.032.078-.078.13-.117.183-.137.333-.306.503-.456a9.948 9.948 0 001.394-1.428c.183-.234.333-.482.49-.73a8.814 8.814 0 00.834-1.662c.104-.26.221-.508.3-.782a9.55 9.55 0 00.404-2.549c0-.032.02-.065.02-.098 0-.006-.007-.013-.007-.013 0-.006.006-.013.006-.02V38.046z"
        />
      </svg>
      <h1
        class="m-0 font-bold text-neutral-900 transition-colors duration-300 dark:text-neutral-50 text-5xl leading-tight"
      >
        John Doe
      </h1>
      <p
        class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 text-base leading-6"
      >
        John Doe is a software engineer.
      </p>
      <div
        class="h-10 space-y-2 text-center"
      />
      <div
        class="flex justify-center"
      >
        <button
          class="flex h-11 cursor-pointer items-center rounded-md border-0 bg-primary-600 px-6 text-sm font-medium leading-none text-neutral-50 transition-colors duration-300 hover:bg-primary-700 active:bg-primary-800 umami--click--resume-en"
        >
          <svg
            class="mr-2 text-lg"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 7v8a2 2 0 002 2h6M8 7V5a2 2 0 012-2h4.586a1 1 0 01.707.293l4.414 4.414a1 1 0 01.293.707V15a2 2 0 01-2 2h-2M8 7H6a2 2 0 00-2 2v10a2 2 0 002 2h8a2 2 0 002-2v-2"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          resume
        </button>
      </div>
      <div
        data-testid="skills"
      />
      <div
        class="mx-auto mt-2 md:mt-6"
      >
        <a
          aria-hidden="true"
          href="/"
        >
          <svg
            class="h-6 w-6 animate-bounce"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 4a.5.5 0 0 1 .5.5v5.793l2.146-2.147a.5.5 0 0 1 .708.708l-3 3a.5.5 0 0 1-.708 0l-3-3a.5.5 0 1 1 .708-.708L7.5 10.293V4.5A.5.5 0 0 1 8 4z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </div>
    </div>
  </header>
  <div
    class="relative h-screen"
  >
    <img
      alt="John Doe"
    />
    <div
      class="absolute inset-0 bg-gradient-to-t from-neutral-50 to-transparent dark:from-neutral-900"
    />
  </div>
</div>
`;
